examples/src/main/java/org/apache/ignite/examples/ml/tutorial/hyperparametertuning/Step_1_RandomSearch.java

```diff
@@ -0,0 +1,198 @@
+ * is taken as a surrogate of the performance of the model trained on all cases.</p>
+ * <p>
+ * All scenarios are described there: https://sebastianraschka.com/faq/docs/evaluate-a-model.html</p>
+ */
+public class Step_1_RandomSearch {
+    /** Run example. */
+    public static void main(String[] args) {
+        System.out.println();
+        System.out.println(">>> Tutorial step 8 (cross-validation with param grid) example started.");
+
+        try (Ignite ignite = Ignition.start("examples/config/example-ignite.xml")) {
+            try {
+                IgniteCache<Integer, Vector> dataCache = TitanicUtils.readPassengers(ignite);
+
+                // Extracts "pclass", "sibsp", "parch", "sex", "embarked", "age", "fare".
+                final Vectorizer<Integer, Vector, Integer, Double> vectorizer
+                    = new DummyVectorizer<Integer>(0, 3, 4, 5, 6, 8, 10).labeled(1);
+
+                TrainTestSplit<Integer, Vector> split = new TrainTestDatasetSplitter<Integer, Vector>()
+                    .split(0.75);
+
+                Preprocessor<Integer, Vector> strEncoderPreprocessor = new EncoderTrainer<Integer, Vector>()
+                    .withEncoderType(EncoderType.STRING_ENCODER)
+                    .withEncodedFeature(1)
+                    .withEncodedFeature(6) // <--- Changed index here.
+                    .fit(ignite,
+                        dataCache,
+                        vectorizer
+                    );
+
+                Preprocessor<Integer, Vector> imputingPreprocessor = new ImputerTrainer<Integer, Vector>()
+                    .fit(ignite,
+                        dataCache,
+                        strEncoderPreprocessor
+                    );
+
+                Preprocessor<Integer, Vector> minMaxScalerPreprocessor = new MinMaxScalerTrainer<Integer, Vector>()
+                    .fit(
+                        ignite,
+                        dataCache,
+                        imputingPreprocessor
+                    );
+
+                NormalizationTrainer<Integer, Vector> normalizationTrainer = new NormalizationTrainer<Integer, Vector>()
+                    .withP(1);
+
+                Preprocessor<Integer, Vector> normalizationPreprocessor = normalizationTrainer
+                    .fit(
+                        ignite,
+                        dataCache,
+                        minMaxScalerPreprocessor
+                    );
+
+                // Tune hyperparams with K-fold Cross-Validation on the split training set.
+
+                DecisionTreeClassificationTrainer trainerCV = new DecisionTreeClassificationTrainer();
+
+                CrossValidation<DecisionTreeNode, Double, Integer, Vector> scoreCalculator
+                    = new CrossValidation<>();
+
+                ParamGrid paramGrid = new ParamGrid()
+                    .withParameterSearchStrategy(HyperParameterSearchingStrategy.RANDOM_SEARCH)
```

Comment:
IMHO we should use something like strategy builder instead of enums because such iface brings explosion of parameters count and count of validators in ParamGrid

Reply:
Moreover as user in this case I should remember what type of parameters belongs to selected strategy. But I have very short memory :-)